arangod/RocksDBEngine/Methods/RocksDBTrxMethods.cpp

```diff
@@ -295,8 +295,9 @@ bool RocksDBTrxMethods::iteratorMustCheckBounds(
           _readWriteBatch->GetWriteBatch()->GetDataSize() > 0);
 }
 
-void RocksDBTrxMethods::beginQuery(ResourceMonitor* resourceMonitor,
-                                   bool isModificationQuery) {
+void RocksDBTrxMethods::beginQuery(
+    std::shared_ptr<ResourceMonitor> resourceMonitor,
+    bool isModificationQuery) {
   // report to parent
   RocksDBTrxBaseMethods::beginQuery(resourceMonitor, isModificationQuery);
```

Comment:
not important: resource monitor could be moved here.
```suggestion
  RocksDBTrxBaseMethods::beginQuery(std::move(resourceMonitor), isModificationQuery);
```